src/app/home/recent-spaces-widget/recent-spaces-widget.component.ts

```diff
@@ -0,0 +1,82 @@
+import {
+  Component,
+  ErrorHandler,
+  Input,
+  OnInit
+} from '@angular/core';
+import { Router } from '@angular/router';
+import { uniqWith } from 'lodash';
+import {
+  Observable,
+  ReplaySubject,
+  Subject
+} from 'rxjs';
+
+import {
+  Broadcaster,
+  Logger
+} from 'ngx-base';
+import {
+  Context,
+  Contexts,
+  Space,
+  Spaces,
+  SpaceService
+} from 'ngx-fabric8-wit';
+
+@Component({
+  selector: 'fabric8-recent-spaces-widget',
+  templateUrl: './recent-spaces-widget.component.html'
+})
+export class RecentSpacesWidget implements OnInit {
+
+  @Input() cardSizeClass: string;
+
+  readonly userHasSpaces: Subject<boolean> = new ReplaySubject<boolean>(1);
+  readonly recentSpaces: Observable<Space[]>;
+
+  constructor(
+    private contexts: Contexts,
+    private spaces: Spaces,
+    private spaceService: SpaceService,
+    private router: Router,
+    private errorHandler: ErrorHandler,
+    private broadcaster: Broadcaster,
+    private logger: Logger
+  ) {
+    this.recentSpaces = this.spaces.recent
+      .map((spaces: Space[]): Space[] => uniqWith(spaces, RecentSpacesWidget.compareSpaces));
+  }
+
+  ngOnInit(): void {
+    (this.router.url.startsWith('/_home') ? this.contexts.default : this.contexts.current)
```

Comment:
`git blame` points to this commit [0] as the origin of this bit of logic, with the last small revision to it being [1].

I'm not exactly sure what "menus" are being referenced here. The `Context` is used just to grab the owning user's username and use that for the `SpaceService` query. I'm not sure why this widget would need to be able to display recent spaces according to various different contexts that a user could be looking at. Since we are currently only displaying it on the home page, I could eliminate this logic and just make the assumption that we're on the homepage and use the default context.

[0] https://github.com/fabric8-ui/fabric8-ui/commit/698ce51c3f193e5c30bffbe949dd779f5a3df4cb

[1] https://github.com/fabric8-ui/fabric8-ui/commit/493516db01a0f68836aa7968643311a4c11e79b2